packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -739,7 +836,7 @@ class ContainerLayer extends Layer {
       if (childOffset == Offset.zero) {
         child._addToSceneWithRetainedRendering(builder);
       } else {
-        child.addToScene(builder, childOffset);
+        child.engineLayer = child.addToScene(builder, childOffset);
```

Comment:
seems weird for the parent to be setting engileLayer on the child, especially when setting it to a value that the child returned. Who owns the engineLayer?

Reply:
This is not entirely new. Callers of `addToScene` are already expected to set the `_engineLayer`  field. Here's an example in today's Flutter:

https://github.com/flutter/flutter/blob/69248258aff8ee89aacb4bc1bc211db7577b46a0/packages/flutter/lib/src/rendering/layer.dart#L176

What I'm doing in this PR is teaching more places to remember the engine layers they generated. Initially I changed it so that the layers manage their own engine layers, but then reverted it (see https://github.com/flutter/flutter/pull/36402/commits/2f7aa9ea66454ea2ab0de508d9f143a7113b5f88), deciding that breaking the API may not be worth it. However, I do prefer that layers own their engine layers and their lifecycle. I'd be on board with changing it. At this point I'd prefer doing it in a separate PR though.

Reply:
https://github.com/flutter/flutter/issues/37805 is created to track the effort to make `addToScene` return void and set the `engineLayer` inside `addToScene`.

Reply:
Thanks! I'm totally willing to make that change in a follow-up PR.

Reply:
It being new or not doesn't matter. :-) We want every patch to _reduce_ technical debt.

Happy for this to happen in a later issue. Please make sure to assign it to yourself and set the milestone accordingly (and schedule it accordingly) so that we know it'll happen.

(Maybe it should happen first, before this PR?)